emeval/input/phone_view.py

```diff
@@ -293,36 +293,16 @@ def fill_battery_df(self, storage_key):
                         battery_df["hr"] = (battery_df.ts-r["start_ts"])/3600.0
                     r["battery_df"] = battery_df
 
-    def _read_until_done(self, phone_label, key, start_ts, end_ts):
-        all_done = False
-        location_entries = []
-        curr_start_ts = start_ts
-        prev_retrieved_count = 0
-
-        while not all_done:
-            print("About to retrieve data for %s from %s -> %s" % (phone_label, curr_start_ts, end_ts))
-            curr_location_entries = self.spec_details.retrieve_data_from_server(phone_label, [key], curr_start_ts, end_ts)
-            print("Retrieved %d entries with timestamps %s..." % (len(curr_location_entries), [cle["data"]["ts"] for cle in curr_location_entries[0:10]]))
-            if len(curr_location_entries) == 0 or len(curr_location_entries) == 1:
-                all_done = True
-            else:
-                location_entries.extend(curr_location_entries)
-                new_start_ts = curr_location_entries[-1]["metadata"]["write_ts"]
-                assert new_start_ts > curr_start_ts
-                curr_start_ts = new_start_ts
-                prev_retrieved_count = len(curr_location_entries)
-        return location_entries
-
     def fill_location_df(self, storage_key):
         for phoneOS, phone_map in self.phone_view_map.items():
             print("Processing data for %s phones" % phoneOS)
             for phone_label in phone_map:
                 curr_calibration_ranges = phone_map[phone_label]["{}_ranges".format(storage_key)]
                 for r in curr_calibration_ranges:
-                    r["location_entries"] = self._read_until_done(phone_label,
+                    r["location_entries"] = self.spec_details.read_until_done(phone_label,
```

Comment:
I am not sure this is the right way to refactor this. In particular, I am not sure that this is required for `FileSpecDetails` since file reads do not limit the number of entries returned in a call. This should only really be relevant to `ServerSpecDetails`.

Having said that, we only use `_read_until_done` currently for a subset of the keys.

Two big changes here:
- at least for those keys, the `--key` option also needs to call `_read_until_done`. Note that DRY, and reusing the SpecDetails in the script, will make this trivial.
- while `_read_until_done` is only *required* for a subset of keys, it should work for all keys. After committing everything else, I would explore with changing the implementation of `retrieve_data` for `ServerSpecDetails` to just always use `_read_until_done`. That simplifies the phone view code, and also ensures that we don't try batching for the `FileSpecDetails`  where it is not necessary.